nilearn/plotting/img_plotting.py

```diff
@@ -556,6 +556,168 @@ def plot_roi(roi_img, bg_img=MNI152TEMPLATE, cut_coords=None,
+            Matplotlib figure used or its number. If None is given, a
+            new figure is created.
+        axes : matplotlib axes or 4 tuple of float: (xmin, ymin, width, height), optional
+            The axes, or the coordinates, in matplotlib figure space,
+            of the axes used to display the plot. If None, the complete
+            figure is used.
+        title : string, optional
+            The title displayed on the figure.
+        annotate: boolean, optional
+            If annotate is True, positions and left/right annotation
+            are added to the plot.
+        draw_cross: boolean, optional
+            If draw_cross is True, a cross is drawn on the plot to
+            indicate the cut plosition.
+        black_bg: boolean, optional
+            If True, the background of the image is set to be black. If
+            you wish to save figures with a black background, you
+            will need to pass "facecolor='k', edgecolor='k'" to pylab's
+            savefig.
+        cmap: matplotlib colormap, optional
+            The colormap for the atlas maps
+        vmin: float
+            Lower bound for plotting, passed to matplotlib.pyplot.imshow
+        vmax: float
+            Upper bound for plotting, passed to matplotlib.pyplot.imshow
+        alpha: float between 0 and 1
+            Alpha transparency for the brain schematics.
+    """
+    display = plot_anat(anat_img, cut_coords=cut_coords,
+                        output_file=output_file, display_mode=display_mode,
+                        figure=figure, axes=axes, title=title,
+                        threshold=None, annotate=annotate,
+                        draw_cross=draw_cross, black_bg=black_bg,
+                        cmap=plt.cm.gray, **kwargs)
+
+    maps_img = _utils.check_niimg_4d(maps_img)
+    n_maps = maps_img.shape[3]
+    _view_type = set(('auto', 'contours', 'contours_filled', 'continuous'))
+    if isinstance(view_type, _basestring):
+        if view_type not in _view_type:
+            raise TypeError('view_type option should be given '
+                            'either as a "contours" or "contours_filled" '
+                            'or "continuous" ')
+    elif not isinstance(view_type, _basestring):
+        raise TypeError('view_type option should be '
+                        'either as a "contours" or "contours_filled" '
+                        'or "continuous" but you have given %s '
+                        %type(view_type))
+
+    if isinstance(cmap, _basestring):
+        color_map = plt.cm.get_cmap(cmap)
+
+    # Build a custom colormap for displaying contours
+    color_list = color_map(np.linspace(0, 1, n_maps))
+
+    if threshold is None:
+        # it will use default percentage,
+        # for a nicer look avoiding maximum overlaps for visualization
+        threshold = "99.7%"
+
+    if hasattr(threshold, '__iter__') and not isinstance(threshold, _basestring):
+        if len(threshold) != n_maps:
```

Comment:
> BTW, if we do that and an iterator is given, it will explode deep in the code with a weird error I think.

I reckon there should be a catch-all else at the end that raises if threshold is not of one of the accepted types.